microservices/clients/contestproposal-client/src/main/java/org/xcolab/client/proposals/ProposalAttributeClient.java

```diff
@@ -146,6 +133,15 @@ public boolean updateProposalAttribute(ProposalAttribute proposalAttribute) {
                 .execute(), serviceNamespace);
     }
 
+    public List<ProposalAttribute> getAllProposalAttributesByNameAndAdditionalId(long proposalId,
```

Comment:
What is with the naming? In my opinion, this `ByNameAndAdditionalId` is redundant and makes the method name unnecessarily long. It is not like Objective-C, where you need this type of naming. I would prefer `getProposalAttributes` here.

I'd like to hear your opinion on that. And if you do think that it is useful, in which cases do you think we should name the attributes in the method name and in which should we not?

Reply:
I thought about this a little bit and I tend to agree with you about the names. There are still some cases where it's helpful or necessary to list parameters, so I'm not sure what the convention should be.

The problem is when a method has the same signature, but different parameter names. An example could be `getProposalAttribute(long proposalId)` and `getProposalAttribute(long userId)`. This is not allowed, as the signatures are the same. In that case I think you'd need to add the parameter in the name:
```java
getProposalAttributeByProposalId(long proposalId)
getProposalAttributeByUserId(long userId)
```

In this case I think it's fine to call it `getAllProposalAttributes` - I'll go ahead and make that change and we can talk about what the convention should be in our next call.